### PR TITLE
Exit MemMonitor before jniCheckFatalErrorNLS() if not JNICHK_NONFATAL

### PR DESCRIPTION
**Pull request by JasonFengJ9**
This fixes a hang scenario:
One thread holds VM access and attempts to enter `MemMonitor`;
Another thread already entered `MemMonitor`, `exitJavaVM()` due to `fatalError()`, and `acquireExclusiveVMAccessFromExternalThread()` but can't acquire exclusive VM access because last thread holds the VM access which can't proceed either since this thread holds `MemMonitor`, hence deadlock.
This PR exits `MemMonitor` before invoking `jniCheckFatalErrorNLS()/fatalError()` to avoid the hang scenario above.

closes: #10191 

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by JasonFengJ9** [comment]
@gacholio could you please review?

File: runtime/jnichk/jnicmem.c [hunk 1]
```diff
@@ -122,6 +126,9 @@ jniCheckMemoryInit(J9JavaVM* javaVM)
 	PORT_ACCESS_FROM_JAVAVM(javaVM);
 
 	omrthread_monitor_t globalMonitor = omrthread_global_monitor();
+	if (0 == (javaVM->checkJNIData.options & JNICHK_NONFATAL)) {
```
Comment (gacholio) [inline runtime/jnichk/jnicmem.c:129]: Please use the any bits macro here.

Comment (JasonFengJ9) [reply runtime/jnichk/jnicmem.c]: Yeah, replace it with `J9_ARE_NO_BITS_SET`.
Please have another look.

**Comment by gacholio** [comment]
jenkins test sanity,extended win jdk8

**Pull request merged**